monkey/monkey_island/cc/services/authentication_service/flask_resources/agent_otp_login.py

```diff
@@ -26,27 +31,54 @@ def post(self):
         """
         Gets the one-time password from the request,
         and returns an authentication token and a refresh token
+        for a particular Agent
 
+        :param agent_id: The ID of the Agent trying to log in
         :return: Authentication token in the response body
         """
-
         try:
-            cred_dict = json.loads(request.data)
-            otp = cred_dict.get("otp", "")
-            if self._validate_otp(otp):
-                refresh_token = "refreshtoken"
-
-                response = make_response(
-                    {"response": {"user": {ACCESS_TOKEN_KEY_NAME: "supersecrettoken"}}}
+            try:
+                agent_id_argument = request.json["agent_id"]
+                agent_id = AgentID(agent_id_argument)
+            except ValueError as err:
+                return make_response(
+                    f'Invalid agent ID "{agent_id_argument}": {err}', HTTPStatus.BAD_REQUEST
                 )
-                response = add_refresh_token_to_response(response, refresh_token)
-
-                return response
-
-        except Exception:
-            pass
 
-        return responses.make_response_to_invalid_request()
+            try:
+                otp_argument = request.json["otp"]
+                otp = OTP(otp_argument)
+            except ValueError as err:
+                return make_response(f'Invalid OTP "{otp_argument}": {err}', HTTPStatus.BAD_REQUEST)
+        except KeyError as err:
+            return make_response(f"Missing argument {err}", HTTPStatus.BAD_REQUEST)
+        except TypeError:
+            return make_response("Could not parse the login request", HTTPStatus.BAD_REQUEST)
+
```

Comment:
TODO: Verify that AgentID exists in the agent repository.

Reply:
We can't do that because Agent registration happens after the login.